dashboard/app/models/plc/enrollment_unit_assignment.rb

```diff
@@ -36,6 +36,9 @@ def enroll_user_in_unit_with_learning_modules(learning_modules)
     transaction do
       plc_module_assignments.destroy_all
 
+      # Make sure the required learning modules are included here
+      learning_modules |=  self.plc_course_unit.plc_learning_modules.where(required: true)
```

Comment:
Again, I wonder if this is something we should bake into the `EnrollmentUnitAssignment` (eager) vs. construct dynamically and allow required modules to be added later (lazy).